File: apprain_barcode_scanner_ios/ios/Classes/ScannerPlatformViewFactory.swift
import Flutter
import UIKit

/// Factory that creates ScannerPlatformView instances for Flutter embedding.
class ScannerPlatformViewFactory: NSObject, FlutterPlatformViewFactory {

    private let messenger: FlutterBinaryMessenger
    private let scanEventChannel: FlutterEventChannel
    private let cameraEventChannel: FlutterEventChannel

    private(set) var activeView: ScannerPlatformView?

    init(
        messenger: FlutterBinaryMessenger,
        scanEventChannel: FlutterEventChannel,
        cameraEventChannel: FlutterEventChannel
    ) {
        self.messenger = messenger
        self.scanEventChannel = scanEventChannel
        self.cameraEventChannel = cameraEventChannel
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        let creationParams = args as? [String: Any]
        let view = ScannerPlatformView(
            frame: frame,
            viewId: viewId,
            creationParams: creationParams,
            scanEventChannel: scanEventChannel,
            cameraEventChannel: cameraEventChannel
        )
        activeView = view
        return view
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }

    func dispose() {
        activeView?.disposeResources()
        activeView = nil
    }
}

File: apprain_barcode_scanner_android/android/src/main/kotlin/com/apprain/barcode_scanner/ScannerPlatformViewFactory.kt
package com.apprain.barcode_scanner

import android.app.Activity
import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

/**
 * Factory that creates [ScannerPlatformView] instances.
 *
 * Each Flutter widget that requests the view type "com.apprain.barcode_scanner/scanner_view"
 * will get a new [ScannerPlatformView] backed by CameraX + ML Kit.
 */
class ScannerPlatformViewFactory(
    private val messenger: BinaryMessenger,
    private val scanEventChannel: EventChannel,
    private val cameraEventChannel: EventChannel,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    var activity: Activity? = null
    var activeView: ScannerPlatformView? = null
        private set

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val creationParams = args as? Map<*, *>
        val view = ScannerPlatformView(
            context = context,
            activity = activity,
            viewId = viewId,
            creationParams = creationParams,
            scanEventChannel = scanEventChannel,
            cameraEventChannel = cameraEventChannel,
        )
        activeView = view
        return view
    }

    fun dispose() {
        activeView?.disposeResources()
        activeView = null
    }
}
